csp_billing_adapter_microsoft/plugin.py

```diff
@@ -71,6 +70,102 @@ def meter_billing(
     is attempted 3 times before re-raising the exception to
     calling scope.
     """
+    metered_billing_data = {"request": []}
```

Comment:
Note that you could just declare a variable, e.g. request, here, and use that until you get to the check for request being empty below, and then construct the relevant dictionary with a request entry inside that block...

Reply:
i cleaned this up , please re-review